e2e-tests/smart-update/run

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash
+
+set -o errexit
+set -o xtrace
+
+IMAGE_PXC=${IMAGE_PXC:-"percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0"}
+IMAGE_TO_UPDATE=${IMAGE_TO_UPDATE:-"perconalab/percona-xtradb-cluster-operator:master-pxc8.0"}
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+API="pxc.percona.com/v1-5-0"
+
+function get_pod_names_images {
+    local cluster=${1}
+    local type=${2:-pxc}
+
+    echo -e $(kubectl_bin get pods -l "app.kubernetes.io/instance=${cluster},app.kubernetes.io/component=${type}" \
+                                   -o jsonpath="{range .items[*]}{.metadata.name}{\",\"}{.spec.containers[?(@.name == \"${type}\")].image}{\"\n\"}{end}")
+}
+
+function get_vs_image {
+    local apply_type=${1}
+    echo $(curl -s https://check.percona.com/versions/v1/pxc-operator/$(echo ${API//pxc.percona.com\/v/} | sed 's/-/\./g')/${apply_type} | jq -r '.versions[].matrix.pxc[].imagePath')
+}
+
+function check_last_pod_to_update {
+    local cluster=${1}
+    local initial_primary=${2}
+    local pxc_size=${3}
+    local target_image=${4}
+    
+    set +x
+    echo -n "Waiting for the last pod to update"
+    until [[ "$(kubectl_bin get pxc "${cluster}" -o jsonpath='{.status.state}')" == "ready" ]]; do
```

Comment:
why we cannot remove the line 33?

Reply:
we can't proceed with update until the cluster is ready state